src/dropdown/dropdown.ts

```diff
@@ -242,4 +253,47 @@ export class NgbDropdown implements OnInit, OnDestroy {
       this._menu.position(this._anchor.anchorEl, this.placement);
     }
   }
+
+  private getActiveElmIndex(list): number {
+    let i = 0;
+    let position = -1;
+    for (let elm of list) {
+      if (elm === document.activeElement) {
+        position = i;
+      }
+      i++;
+    }
+    return position;
+  }
+
+  keyboardEvent($event): boolean {
+    if (!this.isOpen()) {
+      this.open();
+    }
+    let list = this.getMenuElements();
+    let position: number = this.getActiveElmIndex(list);
+    switch ($event.which) {
+      case Key.ArrowDown:
+        position++;
+        break;
+      case Key.ArrowUp:
+        position--;
+        break;
+    }
+    if (position >= list.length) {
```

Comment:
This can be simplified to at least something like:

```ts
case Key.ArrowDown:
        position = Math.min(position + 1, list.length - 1);
        break;
```